File: lib/models/ble_mac_rule.dart
import 'package:flutter_ezw_utils/flutter_ezw_index.dart';

part 'ble_mac_rule.g.dart';

/// 蓝牙MAC地址获取规则
/// - 仅iOS使用，iOS不提供MAC地址，可以通过广播内容获取MAC地址
@JsonSerializable()
class BleMacRule {
  final int startIndex;
  final int endIndex;
  //  - 是否反转
  final bool isReverse;
  BleMacRule(
    this.startIndex,
    this.endIndex, {
    this.isReverse = false,
  });

  factory BleMacRule.fromJson(Map<String, dynamic> json) =>
      _$BleMacRuleFromJson(json);

  Map<String, dynamic> toJson() => _$BleMacRuleToJson(this);
}

File: ios/Classes/ble/models/BleMacRule.swift
//
//  BleMacRule.swift
//  flutter_ezw_ble
//
//  Created by Whiskee on 2025/4/28.
//

/**
 * iOS 广播 MAC 解析规则。
 *
 * iOS CoreBluetooth 不暴露真实 BLE MAC，因此业务需要从 manufacturer data 中按范围截取并
 * 可选反转字节序得到展示/匹配用 MAC。
 */
struct BleMacRule: Codable {
    /// 截取起点，包含该下标。
    let startIndex: Int
    /// 截取终点，不包含该下标。
    let endIndex: Int
    /// 是否反转截取后的字节序。
    var isReverse: Bool = false
}
